Plugin/Block/Toolbar.php

```diff
@@ -0,0 +1,172 @@
+use Magento\Framework\Model\ResourceModel\Db\Collection\AbstractCollection;
+
+class Toolbar extends \Magento\Framework\View\Element\Template
+{
+    const SORT_ORDER_DESC = 'DESC';
+
+    /**  @var StoreManagerInterface */
+    private $storeManager;
+
+    /** @var NostoHelperData */
+    private $nostoHelperData;
+
+    /** @var NostoHelperAccount */
+    private $nostoHelperAccount;
+
+    /**  @var CategoryBuilder */
+    private $categoryBuilder;
+
+    /** @var Registry */
+    private $registry;
+
+    /** @var CookieManagerInterface */
+    private $cookieManager;
+
+    /** @var CategoryRecommendation */
+    private $categoryRecommendaiton;
+
+    /** @var AbstractCollection */
+    protected $_collection = null;
+
+    /**
+     * Toolbar constructor.
+     * @param Context $context
+     * @param NostoHelperData $nostoHelperData
+     * @param NostoHelperAccount $nostoHelperAccount
+     * @param StoreManagerInterface $storeManager
+     * @param CategoryBuilder $builder
+     * @param CategoryRecommendation $categoryRecommendaiton
+     * @param CookieManagerInterface $cookieManager
+     * @param Registry $registry
+     * @param array $data
+     */
+    public function __construct(
+        Context $context,
+        NostoHelperData $nostoHelperData,
+        NostoHelperAccount $nostoHelperAccount,
+        StoreManagerInterface $storeManager,
+        CategoryBuilder $builder,
+        CategoryRecommendation $categoryRecommendaiton,
+        CookieManagerInterface $cookieManager,
+        Registry $registry,
+        array $data = []
+    ) {
+        $this->nostoHelperData = $nostoHelperData;
+        $this->nostoHelperAccount = $nostoHelperAccount;
+        $this->categoryBuilder = $builder;
+        $this->storeManager = $storeManager;
+        $this->cookieManager = $cookieManager;
+        $this->categoryRecommendaiton = $categoryRecommendaiton;
+        $this->registry = $registry;
+        parent::__construct($context, $data);
+    }
+
+    /**
+     * Plugin - Used to modify default Sort By filters
+     *
+     * @param \Magento\Catalog\Block\Product\ProductList\Toolbar $subject
+     * @param null $result
+     * @param \Magento\Framework\Data\Collection $collection
+     * @return Toolbar
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function afterSetCollection(
+        MagentoToolbar $subject,
+        $result,
+        $collection
+    ) {
+        $this->_collection = $collection;
+        $store = $this->storeManager->getStore();
+        $currentOrder = $subject->getCurrentOrder();
+        if (
+            $currentOrder !== Config::NOSTO_PERSONALIZED_KEY
+            && $currentOrder !== Config::NOSTO_TOPLIST_KEY
+            || !$this->nostoHelperAccount->nostoInstalledAndEnabled($store)
+            || !$this->nostoHelperData->isCategorySortingEnabled($store)
+        ) {
+            return $this;
+        } else {
```

Comment:
You can avoid nesting this else, since you're returning in both cases. Better readability.